### ISSUE TITLE
webhook "prometheusrulemutate.monitoring.coreos.com": certificate error

### ISSUE THREAD

**Issue by tgoehler** [prometheus-community/helm-charts#2261]
I tried to install the helm chart version and ended up with:

```Error: UPGRADE FAILED: failed to create resource: Internal error occurred: failed calling webhook "prometheusrulemutate.monitoring.coreos.com": Post "https://kube-prometheus-stack-operator.monitoring.svc:443/admission-prometheusrules/validate?timeout=10s": x509: certificate signed by unknown authority (possibly because of "x509: ECDSA verification failure" while trying to verify candidate authority certificate "nil1")```

As the result no servicemonitors exists and no targets are configured in prometheus.

kube-prometheus-stack helm Chart:
version: 35.0.3
appVersion: 0.56.0

k8s version:
v1.21.8

helm Version:
v3.8.0

**Comment by tgoehler** [comment]
I found the root cause.
there where still two objects for the webhooks from an older installation in the default namespace. Normally I install the helm chart into a dedicated namespace, but 
"MutatingWebhookConfiguration" and "ValidatingWebhookConfiguration" objects get installed w caBundle in the default namespace.
So, when cleaning up the cluster before re-installing the helm chart I had to remove them. obviously the new generated certs of the operator didn't mach the caBundle of the old webhook objects.
